docs/usage/types.md

```diff
@@ -707,3 +708,28 @@ Similar validation could be achieved using [`constr(regex=...)`](#constrained-ty
 formatted with a space, the schema would just include the full pattern and the returned value would be a vanilla string.
 
 See [Schema](schema.md) for more details on how the model's schema is generated.
+
+## Allow Arbitrary Types
+
+You can allow arbitrary types using the `arbitrary_types_allowed` config in the [Model Config](model_config.md).
+
+```py
+{!.tmp_examples/types_arbitrary_allowed.py!}
+```
+_(This script is complete, it should run "as is")_
+
+## Generic Classes as Types
+
+!!! warning
+    This is an advanced technique that you might not need in the beginning. In most of the cases you will probably be fine with standard *pydantic* models.
+
+You can use [Generic Classes](https://mypy.readthedocs.io/en/latest/generics.html) as field types and perform custom validation based on the "type parameters" (or sub-types) with `__get_validators__`.
```

Comment:
would it be less confusing to link to https://docs.python.org/3/library/typing.html#typing.Generic? This makes it look a bit like this is a mypy specific thing.

Reply:
Sure thing.